internal/utils/retry.go

```diff
@@ -39,6 +41,12 @@ func (r *Retry) ExecWithRetries(ctx context.Context) *RetryContext {
 	for !retryCtx.Success {
 		retryCtx.RetryCount++
 		if err := r.RetryFunc(); err != nil {
+			if _, ok := err.(*nrErrors.PaymentRequiredError); ok {
```

Comment:
@sanderblue I would expect a test to validate we don't retry for paywall errors.

Reply:
Added